app/src/main/java/org/listenbrainz/android/repository/feed/FeedRepositoryImpl.kt

```diff
@@ -15,12 +15,14 @@ class FeedRepositoryImpl @Inject constructor(
 ) : FeedRepository {
     
     override suspend fun getFeedEvents(
-        username: String,
+        username: String?,
         maxTs: Int?,
         minTs: Int?,
         count: Int
     ) : Resource<FeedData> =
         runCatching {
+            if (username.isNullOrEmpty()) return@runCatching Resource.failure(error = ResponseError.AUTH_HEADER_NOT_FOUND)
```

Comment:
Perform this check before and don't include this in the repos

Reply:
Its just better to have these checks here because then i dont have to repeat this in 10 different view models. A bit anti-societal but i think im done with google's guildlines.

Reply:
Okay tbh I don't oppose this. Let's let it be like this